src/components/mark/mark.styles.ts

```diff
@@ -22,5 +28,25 @@ export const euiMarkStyles = ({ euiTheme, colorMode }: UseEuiTheme) => {
     // Override the browser's black color.
     // Can't use 'inherit' because the text to background color contrast may not be sufficient
     color: ${euiTheme.colors.text};
+
+    // https://seanconnolly.dev/emotion-conditionals
+    ${hasHighlightText === true &&
+    `
+      &:before,
+      &:after {
+        content: ' [${highlightStart}] ';
+        clip-path: inset(100%);
+        clip: rect(1px, 1px, 1px, 1px);
+        height: 1px;
+        overflow: hidden;
+        position: absolute;
+        white-space: nowrap;
+        width: 1px;
+      }
+
+      &:after {
```

Comment:
Nit picky: But I'd rather see the explicit `before` vs `after` text in their own. I find it easier than hunting for cascades.

```suggestion
      &:before,
      &:after {
        clip-path: inset(100%);
        clip: rect(1px, 1px, 1px, 1px);
        height: 1px;
        overflow: hidden;
        position: absolute;
        white-space: nowrap;
        width: 1px;
      }
      
      &:before {
        content: ' [${highlightStart}] ';        
      }

      &:after {
```